### PR TITLE
pushOptionsCountLimit: manage the config setting

### PR DESCRIPTION
**Pull request by migue**
I've created this pull on top of https://github.com/github/spokes-receive-pack/pull/56, so once this is merged I will rebase this on top of the new main branch

The goal here is to deal with the `receive.pushOptionsCountLimit` config setting and make sure we reject any potential push which sends a number of push-options over the configured limit

### CODE REVIEW

File: internal/spokes/spokes.go
Comment (spraints) [inline internal/spokes/spokes.go:606]: Would it make sense to return a count only, rather than the push options? Spokes-receive-pack doesn't use the options, and storing the options in a slice would let the client increase the memory use of spokes-receive-pack somewhat arbitrarily.

**Review by spraints — COMMENTED** [review]
:+1: though I'm going to hold off on approving until after you've rebased.

File: internal/spokes/spokes.go
Comment (migue) [reply internal/spokes/spokes.go]: Change it here 65bdcf392cd8ab5c5dd05ee4ecb1793c85f9ebf2

**Review by spraints — APPROVED** [review]

**Pull request merged**
